Index: plasma-workspace-5.7.2/klipper/CMakeLists.txt
===================================================================
--- plasma-workspace-5.7.2.orig/klipper/CMakeLists.txt
+++ plasma-workspace-5.7.2/klipper/CMakeLists.txt
@@ -19,7 +19,7 @@ set(libklipper_common_SRCS
     clipcommandprocess.cpp
 )
 
-find_package(KF5Prison "1.2.1" EXACT QUIET CONFIG)
+find_package(KF5Prison "1.2.0" QUIET CONFIG)
 set_package_properties(KF5Prison PROPERTIES DESCRIPTION "Prison library"
                        URL "http://projects.kde.org/prison"
                        TYPE OPTIONAL
