diff --git a/applets/systemmonitor/memory/metadata.desktop b/applets/systemmonitor/memory/metadata.desktop
index da3f0bd..03f35d4 100644
--- a/applets/systemmonitor/memory/metadata.desktop
+++ b/applets/systemmonitor/memory/metadata.desktop
@@ -12,6 +12,7 @@ Name[da]=Hukommelsesstatus
 Name[de]=Speicherstatus
 Name[el]=Κατάσταση μνήμης
 Name[en_GB]=Memory Status
+Name[eo]=Memora stato
 Name[es]=Estado de la memoria
 Name[et]=Mälu olek
 Name[eu]=Memoria-egoera
diff --git a/containmentactions/applauncher/plasma-containmentactions-applauncher.desktop b/containmentactions/applauncher/plasma-containmentactions-applauncher.desktop
index 11df229..6dbea05 100644
--- a/containmentactions/applauncher/plasma-containmentactions-applauncher.desktop
+++ b/containmentactions/applauncher/plasma-containmentactions-applauncher.desktop
@@ -100,6 +100,7 @@ Comment[da]=Simpel startmenu
 Comment[de]=Einfacher Anwendungsstarter
 Comment[el]=Απλή εκτέλεση εφαρμογών
 Comment[en_GB]=Simple application launcher
+Comment[eo]=Simpla plikaĵolanĉilo
 Comment[es]=Lanzador de aplicaciones sencillo
 Comment[et]=Lihtne rakenduste käivitaja
 Comment[eu]=Aplikazio-abiarazle soila
diff --git a/krunner/view.cpp b/krunner/view.cpp
index b1bee97..48bab0c 100644
--- a/krunner/view.cpp
+++ b/krunner/view.cpp
@@ -256,6 +256,13 @@ bool View::event(QEvent *event)
     return retval;
 }
 
+void View::resizeEvent(QResizeEvent *event)
+{
+    if (event->oldSize().width() != event->size().width()) {
+        positionOnScreen();
+    }
+}
+
 void View::showEvent(QShowEvent *event)
 {
     KWindowSystem::setOnAllDesktops(winId(), true);
diff --git a/krunner/view.h b/krunner/view.h
index 7314a21..792088e 100644
--- a/krunner/view.h
+++ b/krunner/view.h
@@ -64,6 +64,7 @@ Q_SIGNALS:
 
 protected:
     bool event(QEvent* event) override;
+    void resizeEvent(QResizeEvent *event) override;
     void showEvent(QShowEvent *event) override;
 
 public Q_SLOTS:
diff --git a/libtaskmanager/taskfilterproxymodel.cpp b/libtaskmanager/taskfilterproxymodel.cpp
index 79b902e..803db21 100644
--- a/libtaskmanager/taskfilterproxymodel.cpp
+++ b/libtaskmanager/taskfilterproxymodel.cpp
@@ -288,7 +288,8 @@ bool TaskFilterProxyModel::filterAcceptsRow(int sourceRow, const QModelIndex &so
 
     // Filter by virtual desktop.
     if (d->filterByVirtualDesktop && d->virtualDesktop != 0) {
-        if (!sourceIdx.data(AbstractTasksModel::IsOnAllVirtualDesktops).toBool()) {
+        if (!sourceIdx.data(AbstractTasksModel::IsOnAllVirtualDesktops).toBool()
+            && !sourceIdx.data(AbstractTasksModel::IsDemandingAttention).toBool()) {
             const QVariant &virtualDesktop = sourceIdx.data(AbstractTasksModel::VirtualDesktop);
 
             if (!virtualDesktop.isNull()) {
diff --git a/menu/desktop/kf5-system.directory b/menu/desktop/kf5-system.directory
index 0ef84ef..10658ca 100644
--- a/menu/desktop/kf5-system.directory
+++ b/menu/desktop/kf5-system.directory
@@ -13,6 +13,7 @@ Name[da]=System
 Name[de]=System
 Name[el]=Σύστημα
 Name[en_GB]=System
+Name[eo]=Sistemo
 Name[es]=Sistema
 Name[et]=Süsteem
 Name[fi]=Järjestelmä
diff --git a/menu/desktop/kf5-utilities-file.directory b/menu/desktop/kf5-utilities-file.directory
index 220eab0..ec2e150 100644
--- a/menu/desktop/kf5-utilities-file.directory
+++ b/menu/desktop/kf5-utilities-file.directory
@@ -13,6 +13,7 @@ Name[da]=Fil
 Name[de]=Datei
 Name[el]=Αρχείο
 Name[en_GB]=File
+Name[eo]=Dosiero
 Name[es]=Archivo
 Name[et]=Fail
 Name[fi]=Tiedosto
@@ -63,6 +64,7 @@ Comment[da]=Fil
 Comment[de]=Datei
 Comment[el]=Αρχείο
 Comment[en_GB]=File
+Comment[eo]=Dosiero
 Comment[es]=Archivo
 Comment[et]=Fail
 Comment[fi]=Tiedosto
diff --git a/menu/desktop/kf5-utilities.directory b/menu/desktop/kf5-utilities.directory
index 55a90b5..d22bea5 100644
--- a/menu/desktop/kf5-utilities.directory
+++ b/menu/desktop/kf5-utilities.directory
@@ -12,6 +12,7 @@ Name[da]=Værktøjer
 Name[de]=Dienstprogramme
 Name[el]=Εργαλεία
 Name[en_GB]=Utilities
+Name[eo]=Iloj
 Name[es]=Utilidades
 Name[et]=Tööriistad
 Name[fi]=Apuohjelmat
@@ -61,6 +62,7 @@ Comment[da]=Værktøjer
 Comment[de]=Dienstprogramme
 Comment[el]=Εργαλεία
 Comment[en_GB]=Utilities
+Comment[eo]=Iloj
 Comment[es]=Utilidades
 Comment[et]=Tööriistad
 Comment[fi]=Apuohjelmat
diff --git a/phonon/platform_kde/phonon.notifyrc b/phonon/platform_kde/phonon.notifyrc
index 7e4ff08..200e40e 100644
--- a/phonon/platform_kde/phonon.notifyrc
+++ b/phonon/platform_kde/phonon.notifyrc
@@ -79,6 +79,7 @@ Name[da]=Program
 Name[de]=Anwendung
 Name[el]=Εφαρμογή
 Name[en_GB]=Application
+Name[eo]=Aplikaĵo
 Name[es]=Aplicación
 Name[et]=Rakendus
 Name[fi]=Sovellus
diff --git a/runners/baloo/plasma-runner-baloosearch.desktop b/runners/baloo/plasma-runner-baloosearch.desktop
index a2d98a3..6b390bf 100644
--- a/runners/baloo/plasma-runner-baloosearch.desktop
+++ b/runners/baloo/plasma-runner-baloosearch.desktop
@@ -52,6 +52,7 @@ Comment[de]=Durchsucht Dateien, E-Mails und Kontakte
 Comment[el]=Αναζήτηση σε αρχεία, μηνύματα ηλ. ταχυδρομείου και επαφές
 Comment[en_GB]=Searches through files, emails and contacts
 Comment[es]=Busca en archivos, correo electrónico y contactos
+Comment[et]=Otsimine failides, e-kirjades ja kontaktides
 Comment[fi]=Etsii tiedostoista, sähköposteista ja yhteystiedoista
 Comment[gl]=Busca nos ficheiros, correos e contactos.
 Comment[is]=Leitar í skrám, tölvupósti og tengiliðum
diff --git a/shell/CMakeLists.txt b/shell/CMakeLists.txt
index d323e48..2920187 100644
--- a/shell/CMakeLists.txt
+++ b/shell/CMakeLists.txt
@@ -111,4 +111,3 @@ install(FILES
    DESTINATION ${KDE_INSTALL_KSERVICETYPES5DIR})
 
 add_subdirectory(packageplugins)
-add_subdirectory(tests)
diff --git a/shell/desktopview.cpp b/shell/desktopview.cpp
index 509184d..83866dc 100644
--- a/shell/desktopview.cpp
+++ b/shell/desktopview.cpp
@@ -193,7 +193,7 @@ bool DesktopView::event(QEvent *e)
     } else if (e->type() == QEvent::FocusIn) {     //FIXME: this should *not* be needed
         ensureWindowType();
 
-    } else if (e->type() == QEvent::FocusOut) {
+    } else if (e->type() == QEvent::FocusOut && containment()) {
         QObject *graphicObject = containment()->property("_plasma_graphicObject").value<QObject *>();
         if (graphicObject) {
             graphicObject->setProperty("focus", false);
diff --git a/shell/packageplugins/lookandfeel/plasma-packagestructure-lookandfeel.desktop b/shell/packageplugins/lookandfeel/plasma-packagestructure-lookandfeel.desktop
index 39ec0f0..e474317 100644
--- a/shell/packageplugins/lookandfeel/plasma-packagestructure-lookandfeel.desktop
+++ b/shell/packageplugins/lookandfeel/plasma-packagestructure-lookandfeel.desktop
@@ -16,7 +16,7 @@ Name[eu]=Itxura eta Izaera
 Name[fi]=Ulkoasu ja tuntuma
 Name[fr]=Apparence
 Name[gl]=Aparencia e comportamento
-Name[he]=מראה והרגשה
+Name[he]=מראה ותחושה
 Name[hu]=Megjelenés
 Name[ia]=Semblantia
 Name[id]=Tampilan dan Rasa
diff --git a/shell/scripting/scriptengine.cpp b/shell/scripting/scriptengine.cpp
index b066c6a..d04dad3 100644
--- a/shell/scripting/scriptengine.cpp
+++ b/shell/scripting/scriptengine.cpp
@@ -211,16 +211,18 @@ QScriptValue ScriptEngine::createActivity(QScriptContext *context, QScriptEngine
 
     qDebug() << "Setting default Containment plugin:" << plugin;
 
-    if (plugin.isEmpty() || plugin == QLatin1String("undefined")) {
-        KConfigGroup shellCfg = KConfigGroup(KSharedConfig::openConfig(env->m_corona->package().filePath("defaults")), "Desktop");
-        plugin = shellCfg.readEntry("Containment", "org.kde.desktopcontainment");
-    }
-
     ShellCorona *sc = qobject_cast<ShellCorona *>(env->m_corona);
     StandaloneAppCorona *ac = qobject_cast<StandaloneAppCorona *>(env->m_corona);
     if (sc) {
+        if (plugin.isEmpty() || plugin == QLatin1String("undefined")) {
+            plugin = sc->defaultContainmentPlugin();
+        }
         sc->insertActivity(id, plugin);
     } else if (ac) {
+        if (plugin.isEmpty() || plugin == QLatin1String("undefined")) {
+            KConfigGroup shellCfg = KConfigGroup(KSharedConfig::openConfig(env->m_corona->package().filePath("defaults")), "Desktop");
+            plugin = shellCfg.readEntry("Containment", "org.kde.desktopcontainment");
+        }
         ac->insertActivity(id, plugin);
     }
 
@@ -863,7 +865,8 @@ QScriptValue ScriptEngine::desktops(QScriptContext *context, QScriptEngine *engi
     int count = 0;
 
     foreach (Plasma::Containment *c, env->corona()->containments()) {
-        if (!isPanel(c)) {
+        //make really sure we get actual desktops, so check for a non empty activty id
+        if (!isPanel(c) && !c->activity().isEmpty()) {
             containments.setProperty(count, env->wrap(c));
             ++count;
         }
diff --git a/shell/shellcorona.cpp b/shell/shellcorona.cpp
index 93a218a..c993eec 100644
--- a/shell/shellcorona.cpp
+++ b/shell/shellcorona.cpp
@@ -117,6 +117,8 @@ ShellCorona::ShellCorona(QObject *parent)
     connect(&m_reconsiderOutputsTimer, &QTimer::timeout, this, &ShellCorona::reconsiderOutputs);
 
     m_desktopDefaultsConfig = KConfigGroup(KSharedConfig::openConfig(package().filePath("defaults")), "Desktop");
+    m_lnfDefaultsConfig = KConfigGroup(KSharedConfig::openConfig(m_lookAndFeelPackage.filePath("defaults")), "Desktop");
+    m_lnfDefaultsConfig = KConfigGroup(&m_lnfDefaultsConfig, QStringLiteral("org.kde.plasma.desktop"));
 
     new PlasmaShellAdaptor(this);
 
@@ -243,6 +245,8 @@ void ShellCorona::setShell(const QString &shell)
     package.setAllowExternalPaths(true);
     setKPackage(package);
     m_desktopDefaultsConfig = KConfigGroup(KSharedConfig::openConfig(package.filePath("defaults")), "Desktop");
+    m_lnfDefaultsConfig = KConfigGroup(KSharedConfig::openConfig(m_lookAndFeelPackage.filePath("defaults")), "Desktop");
+    m_lnfDefaultsConfig = KConfigGroup(&m_lnfDefaultsConfig, shell);
 
     const QString themeGroupKey = QStringLiteral("Theme");
     const QString themeNameKey = QStringLiteral("name");
@@ -302,7 +306,9 @@ void ShellCorona::setShell(const QString &shell)
 
     connect(m_activityController, &KActivities::Controller::serviceStatusChanged, this, &ShellCorona::load, Qt::UniqueConnection);
 
-    load();
+    if (m_activityController->serviceStatus() == KActivities::Controller::Running) {
+        load();
+    }
 }
 
 QString ShellCorona::shell() const
@@ -330,7 +336,7 @@ static QList<QScreen*> sortOutputs(const QList<QScreen*> &outputs)
 void ShellCorona::load()
 {
     if (m_shell.isEmpty() ||
-        m_activityController->serviceStatus() == KActivities::Controller::Unknown) {
+        m_activityController->serviceStatus() != KActivities::Controller::Running) {
         return;
     }
 
@@ -339,7 +345,12 @@ void ShellCorona::load()
     loadLayout("plasma-" + m_shell + "-appletsrc");
 
     checkActivities();
+
     if (containments().isEmpty()) {
+        // Seems like we never really get to this point since loadLayout already
+        // (virtually) calls loadDefaultLayout if it does not load anything
+        // from the config file. Maybe if the config file is not empty,
+        // but still does not have any containments
         loadDefaultLayout();
         processUpdateScripts();
     } else {
@@ -569,6 +580,14 @@ void ShellCorona::loadDefaultLayout()
         QString code = file.readAll();
         qDebug() << "evaluating startup script:" << script;
 
+        // We need to know which activities are here in order for
+        // the scripting engine to work. activityAdded does not mind
+        // if we pass it the same activity multiple times
+        QStringList existingActivities = m_activityController->activities();
+        foreach (const QString &id, existingActivities) {
+            activityAdded(id);
+        }
+
         WorkspaceScripting::ScriptEngine scriptEngine(this);
 
         connect(&scriptEngine, &WorkspaceScripting::ScriptEngine::printError, this,
@@ -915,7 +934,7 @@ Plasma::Containment *ShellCorona::createContainmentForActivity(const QString& ac
     QString plugin = m_activityContainmentPlugins.value(activity);
 
     if (plugin.isEmpty()) {
-        plugin = m_desktopDefaultsConfig.readEntry("Containment", "org.kde.desktopcontainment");
+        plugin = defaultContainmentPlugin();
     }
 
     Plasma::Containment *containment = containmentForScreen(screenNum, plugin, QVariantList());
@@ -1218,8 +1237,7 @@ void ShellCorona::activityAdded(const QString &id)
         return;
     }
 
-    const QString plugin = m_desktopDefaultsConfig.readEntry("Containment", "org.kde.desktopcontainment");
-    m_activityContainmentPlugins.insert(id, plugin);
+    m_activityContainmentPlugins.insert(id, defaultContainmentPlugin());
 }
 
 void ShellCorona::activityRemoved(const QString &id)
@@ -1697,6 +1715,15 @@ KWayland::Client::PlasmaShell *ShellCorona::waylandPlasmaShellInterface() const
     return m_waylandPlasmaShell;
 }
 
+QString ShellCorona::defaultContainmentPlugin() const
+{
+    QString plugin = m_lnfDefaultsConfig.readEntry("Containment", QString());
+    if (plugin.isEmpty()) {
+        plugin = m_desktopDefaultsConfig.readEntry("Containment", "org.kde.desktopcontainment");
+    }
+    return plugin;
+}
+
 void ShellCorona::updateStruts()
 {
     foreach(PanelView* view, m_panelViews) {
diff --git a/shell/shellcorona.h b/shell/shellcorona.h
index 4a2f7f9..55b14c4 100644
--- a/shell/shellcorona.h
+++ b/shell/shellcorona.h
@@ -106,6 +106,8 @@ public:
 
     KWayland::Client::PlasmaShell *waylandPlasmaShellInterface() const;
 
+    QString defaultContainmentPlugin() const;
+
 protected:
     bool eventFilter(QObject *watched, QEvent *event) override;
 
@@ -213,6 +215,7 @@ private:
     KActivities::Controller *m_activityController;
     QHash<const Plasma::Containment *, PanelView *> m_panelViews;
     KConfigGroup m_desktopDefaultsConfig;
+    KConfigGroup m_lnfDefaultsConfig;
     QList<Plasma::Containment *> m_waitingPanels;
     QHash<QString, QString> m_activityContainmentPlugins;
     QHash<QString, QHash<int, Plasma::Containment *> > m_desktopContainments;
diff --git a/shell/tests/CMakeLists.txt b/shell/tests/CMakeLists.txt
deleted file mode 100644
index 4c948a6..0000000
--- a/shell/tests/CMakeLists.txt
+++ /dev/null
@@ -1,17 +0,0 @@
-function(plasmoidtest plasmoidName)
-    if (Qt5Core_VERSION_STRING VERSION_LESS "5.6.1")
-        return()
-    endif ()
-    configure_file(
-            ${CMAKE_CURRENT_SOURCE_DIR}/testLayout.js.in
-            ${CMAKE_CURRENT_BINARY_DIR}/testLayout-${plasmoidName}.js
-            @ONLY
-    )
-
-    add_test(NAME ${plasmoidName}-test COMMAND
-            plasmashell --test ${CMAKE_CURRENT_BINARY_DIR}/testLayout-plasmoidName.js
-    )
-endfunction()
-
-plasmoidtest(org.kde.plasma.analogclock)
-plasmoidtest(org.kde.plasma.kickoff)
diff --git a/shell/tests/testLayout.js.in b/shell/tests/testLayout.js.in
deleted file mode 100644
index e6002d4..0000000
--- a/shell/tests/testLayout.js.in
+++ /dev/null
@@ -1,14 +0,0 @@
-var appletName = "@plasmoidName@";
-
-var panel = new Panel
-panel.location = "top";
-panel.height = gridUnit * 2
-panel.addWidget(appletName)
-
-var id = createActivity("Test: "+ appletName);
-var desktopsArray = desktopsForActivity(id);
-for( var j in desktopsArray) {
-    var desktop = desktopsArray[j];
-    desktop.wallpaperPlugin = 'org.kde.color';
-    desktop.addWidget(appletName);
-}
diff --git a/templates/ion-dataengine/src/ion-%{APPNAMELC}.desktop b/templates/ion-dataengine/src/ion-%{APPNAMELC}.desktop
index 409a35b..b5d3e32 100644
--- a/templates/ion-dataengine/src/ion-%{APPNAMELC}.desktop
+++ b/templates/ion-dataengine/src/ion-%{APPNAMELC}.desktop
@@ -7,6 +7,7 @@ Name[de]=%{APPNAME}-Wetterdatenanbieter
 Name[el]=%{APPNAME} Πάροχος δεδομένων καιρού
 Name[en_GB]=%{APPNAME} Weather Data Provider
 Name[es]=Proveedor de datos meteorológicos %{APPNAME}
+Name[et]=%{APPNAME} ilmaandmete pakkuja
 Name[fi]=%{APPNAME} -säätietojen tarjoaja
 Name[gl]=Provedor de datos meteorolóxicos %{APPNAME}
 Name[he]=ספק מזג האוויר %{APPNAME}
@@ -37,6 +38,7 @@ Comment[de]=Zugriff auf den Wetterdatenanbieter %{APPNAME}
 Comment[el]=Πρόσβαση στον πάροχο δεδομένων καιρού %{APPNAME}
 Comment[en_GB]=Access to the weather data provider %{APPNAME}
 Comment[es]=Acceso al proveedor de datos meteorológicos %{APPNAME}
+Comment[et]=Ilmaandmete pakkuja  %{APPNAME} andmete kasutamine
 Comment[fi]=Pääsy säätietojen tarjoajaan %{APPNAME}
 Comment[gl]=Acceder ao provedor de datos meteorolóxicos %{APPNAME}.
 Comment[he]=גישה למספק מזג האוויר %{APPNAME} 
