Index: plasma-workspace/shell/shellcorona.cpp
===================================================================
--- plasma-workspace.orig/shell/shellcorona.cpp
+++ plasma-workspace/shell/shellcorona.cpp
@@ -302,9 +302,7 @@ void ShellCorona::setShell(const QString
 
     connect(m_activityController, &KActivities::Controller::serviceStatusChanged, this, &ShellCorona::load, Qt::UniqueConnection);
 
-    if (m_activityController->serviceStatus() == KActivities::Controller::Running) {
-        load();
-    }
+    load();
 }
 
 QString ShellCorona::shell() const
@@ -332,7 +330,7 @@ static QList<QScreen*> sortOutputs(const
 void ShellCorona::load()
 {
     if (m_shell.isEmpty() ||
-        m_activityController->serviceStatus() != KActivities::Controller::Running) {
+        m_activityController->serviceStatus() == KActivities::Controller::Unknown) {
         return;
     }
 
